### PR TITLE
workflow clean up

### PR DESCRIPTION
**Pull request by mwdchang**
### Summary
Fix a potential problem where a delete node/edge may get undeleted.

Remove outdated version shim

Add createdBy auditing to better track when nodes/edges are created to better debug synchronization issues

### CODE REVIEW

File: packages/server/src/main/java/software/uncharted/terarium/hmiserver/service/data/WorkflowService.java [hunk 1]
```diff
@@ -130,19 +130,13 @@ public Optional<Workflow> updateAsset(
...
 
-				// FIXME: backwards compatibility for older workflows, remove in a few month.
-				// Aug 2024
-				if (dbNode.getVersion() == null) {
-					dbNode.setVersion(1L);
-					continue;
-				}
-
-				if (dbNode.getVersion().equals(node.getVersion())) {
+				if (dbNode.getIsDeleted() == false && dbNode.getVersion().equals(node.getVersion())) {
```
Comment (Tom-Szendrey) [inline packages/server/src/main/java/software/uncharted/terarium/hmiserver/service/data/WorkflowService.java:139]: nitpick: Could use a comment here

**Review by Tom-Szendrey — APPROVED** [review]

**Review by blanchco — APPROVED** [review]

File: packages/client/hmi-client/src/services/workflow.ts [hunk 2]
```diff
@@ -211,6 +212,13 @@ export class WorkflowWrapper {
 	}
 
 	addNode(op: Operation, pos: Position, options: { size?: OperatorNodeSize; state?: any }) {
+		let currentUserName: string | undefined = '';
+		try {
+			currentUserName = useAuthStore().user?.username;
```
Comment (YohannParis) [inline packages/client/hmi-client/src/services/workflow.ts:217]: Using the initials like we do for the user menu would make a lot more sense.
```suggestion
			currentUserName = useAuthStore().userInitials;
```

**Review by YohannParis — APPROVED** [review]
LGTM

File: packages/server/src/main/java/software/uncharted/terarium/hmiserver/service/data/WorkflowService.java (hunk 1 above)
Comment (mwdchang) [reply packages/server/src/main/java/software/uncharted/terarium/hmiserver/service/data/WorkflowService.java:140]: done

File: packages/client/hmi-client/src/services/workflow.ts (hunk 2 above)
Comment (YohannParis) [reply packages/client/hmi-client/src/services/workflow.ts:217]: If the goal is to keep track of which user and not for displaying, you should use the id.

Comment (mwdchang) [reply packages/client/hmi-client/src/services/workflow.ts:217]: You are not wrong, but between workflowId, nodeId, source, target, active-output, source-port, target-port and a few others, that is a whole lot of uuids to track and having one less helps.

**Pull request merged**